corehq/apps/events/models.py

```diff
@@ -480,6 +481,28 @@ def save(self, *args, **kwargs):
         }
         helper.update(case_data)
 
+    def delete(self, *args, **kwargs):
+        helper = CaseHelper(case_id=self.case_id, domain=self.domain)
+        helper.close()
+
+    def active_event_count(self, *args, **kwargs):
+        """
+        Returns how many events this attendee is currently being tracked in.
+        Specifically, these are events that have not started yet, or are currently in progress.
+        """
+        # Get events that haven't started yet or are in progress
+        today = date.today()
+        events = Event.objects.by_domain(self.domain).filter(
+            Q(start_date__gt=today) | Q(start_date__lte=today, end_date__gte=today)
```

Comment:
Correct me if I'm wrong, but shouldn't we be concerned about events that are either _in progress_ or _done_ (as opposed to _have not started_)?

I'm not sure I understand why we are concerned whether or not the attendee is part of an event that is yet to commence, since the said attendee cannot be tracked so such an event yet?

Reply:
`start_date` is either > today, or it's <= today.
```python
>>> False or (True and 'foo')
'foo'
>>> True or (False and 'foo')
True
```
So you could simplify this filter a little to `Q(start_date__gt=today) | Q(end_date__gte=today)`.

Charl might have a point though, in which case we could just use `filter(start_date__lte=today, end_date__gte=today)`.

Reply:
@Charl1996 That is a good point, I just included events that _have not started_ as a sanity check so that someone does not accidently delete an attendee without being aware that they are currently enlisted for one or more future events. It might make more sense to have this as a warning rather than a blocker though.

Would we want to prevent deleting users that have no _in progress_ events but _completed_ events? This would mean that once an attendee has done any event, they can never be deleted. Or should we only be concerned about preventing deletion of attendees that have _in progress_ events? :thinking:

Reply:
@kaapstorm Ah that makese sense. I'll simplify the query a bit based on your suggestion.

Reply:
@zandre-eng 
> Would we want to prevent deleting users that have no in progress events but completed events?

That depends on what a "tracked attendee" is.

From the ticket:
> Users may only remove attendees up until they have been tracked in an event

I interpret this as "only be able to remove attendees until they have actually attended any event".

> This would mean that once an attendee has done any event, they can never be deleted.

Correct.

I don't think we should be able to remove attendees once they attended any event (as said above), because throwing away an attendee record is throwing away someone's compensation.

Reply:
(PS. it goes without saying that this also applies to events that are in progress, thus "tracked attendee" = attendee that has attended any past event OR attendee that is attending an event that is in progress)

Reply:
Thank you for the clarification @Charl1996, that definitely makes sense. I'll make the necessary changes so that the correct attendees are tracked. :+1: